src/main.py

```diff
@@ -178,6 +185,60 @@ def run_plan_bot(
             skip_tracking=skip_tracking,
         )
 
+    params = {}
+    if comments_progress:
+        newest_comment_id = comments_progress.get("newest")
+        oldest_comment_id = comments_progress.get("oldest")
+
+        if newest_comment_id:
+            params["before"] = latest_comment_id
+
+    # Get comments newer than latest
+    current_newest_comment_id = None
+    for comment in subreddit.comments(params=params):
+        comment = reddit_util.Comment(comment)
+        process_post(
+            comment,
+            plans,
+            posts_db,
+            post_ids_processed,
+            send=send_replies,
+            simulate=simulate_replies,
+            skip_tracking=skip_tracking,
+        )
+        if not current_newest_comment_id:
+            current_newest_comment_id = comment.fullname
+            if not skip_tracking:
+                comments_progress.reference.set(
+                    {"newest": current_latest_comment_id}, merge=True
+                )
+
+        # We don't already have an oldest comment from a previous run so set it here
+        if not skip_tracking and not oldest_comment_id:
+            comments_progress.reference.set({"oldest": comment.fullname}, merge=True)
+
+    # If no newer comments, let's go further in the past
```

Comment:
I'm tempted to say that we should leave out the oldest comment logic, or perhaps move it into a separate script.

Reasoning:

- It's untested, and after it runs once, it'll become a no-op. So it seems very likely to break in the future (perhaps after a refactor) without us noticing. Could therefore result in unexpected behavior down the line
- We wouldn't expect to get a ton of value out of running it repeatedly. Since it'll only get exercised once. And so it would effectively just add a 2 second sleep, because of PRAW rate limit

Reply:
Fair enough. I was on the fence about including this at all. It probably safe to say that we don't "care" about older comments because, presumably, push shift will have handled those already. To keep this simple for now, I'd vote to probably exclude it.